go/mysql/mariadb_gtid.go

```diff
@@ -200,6 +210,44 @@ func (gtidSet MariadbGTIDSet) AddGTID(other GTID) GTIDSet {
 	return append(gtidSet, mdbOther)
 }
 
+// Union implements GTIDSet.Union(). This is a pure method, and does not mutate the caller.
+func (gtidSet MariadbGTIDSet) Union(other GTIDSet) GTIDSet {
+	if gtidSet == nil || other == nil {
+		return gtidSet
```

Comment:
shouldn't we return the non-nil set if either is nil?

Reply:
If the receiver is nil, then we return nil. Otherwise, if gtidSet is non-nil, but other is nil, we return the receiver. Are you suggesting that if the receiver is nil we generate a valid empty set? I think in this case this is an invalid method call, and usually in Go we would return nil?

Reply:
If we call Union on a nil receiver that would be an error anyway, so the nil check on the receiver can be removed.

Reply:
@deepthi so your feeling is that a nil receiver should cause a panic. In that case how about a non-nil receiver but `other` is nil? In that case I would imagine we should return the receiver, because union between something + nothing = something.

Reply:
I was looking at MariadbGTIDSet as a struct, I see now that it is a slice. I agree with @systay on this.

Reply:
@systay, @deepthi and I just chatted about this on slack. I think we are in agreement with you now. I'll update the code to reflect this new logic.

Reply:
Updated: https://github.com/vitessio/vitess/pull/6180/commits/aab2a6547eddcef8318d715433785bb502c3c295.